runtime/doc/autocmd.txt

```diff
@@ -859,6 +860,10 @@ TabEnter			Just after entering a tab page. |tab-page|
 TabLeave			Just before leaving a tab page. |tab-page|
 				A WinLeave event will have been triggered
 				first.
+							*TabNew*
+TabNew                          Before entering a new tab page. |tab-page|
+                                After WinEnter and before TabEnter.
+
```

Comment:
> WinEnter and before TabEnter

So the order is: 

    WinEnter > TabNew > TabNewEntered > TabEnter

can you document that and add a test for it?